### PR TITLE
BUG: Series(range_obj_outside_i8_bounds)

### LINKED ISSUE #30173: BUG: Series(range(...)) fails when range contains values not supported by int64
#### Code Sample, a copy-pastable example if possible

The following fails on `master`:

```python
In [1]: import pandas as pd; pd.__version__
Out[1]: '0.26.0.dev0+1255.gb9f26e21e'

In [2]: pd.Series(range(2**63, 2**63 + 4))
---------------------------------------------------------------------------
OverflowError: Python int too large to convert to C long

In [3]: pd.Series(range(2**73, 2**73 + 4))
---------------------------------------------------------------------------
OverflowError: Python int too large to convert to C long
```

But is successful when doing an intermediate conversion to `list`:
```python
In [4]: pd.Series(list(range(2**63, 2**63 + 4)))
Out[4]: 
0    9223372036854775808
1    9223372036854775809
2    9223372036854775810
3    9223372036854775811
dtype: uint64

In [5]: pd.Series(list(range(2**73, 2**73 + 4)))
Out[5]: 
0    9444732965739290427392
1    9444732965739290427393
2    9444732965739290427394
3    9444732965739290427395
dtype: object
```
#### Problem description

`Series(range(...))` fails when values in the `range` are not supported by `int64`, but construction is successful for an equivalent `list`.

#### Expected Output

I'd expect `In [2]`/`In [3]` to be successful and produce output equivalent to `Out[4]`/`Out[5]`.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : b9f26e21eed1c2664f69475df798fc125e6a6d87
python           : 3.7.3.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 18.6.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.26.0.dev0+1255.gb9f26e21e
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 41.6.0.post20191030
Cython           : 0.29.13
pytest           : 4.6.2
hypothesis       : 4.23.6
sphinx           : 1.8.5
blosc            : None
feather          : None
xlsxwriter       : 1.1.8
lxml.etree       : 4.3.3
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: None
bs4              : 4.7.1
bottleneck       : 1.2.1
fastparquet      : 0.3.0
gcsfs            : None
lxml.etree       : 4.3.3
matplotlib       : 3.1.0
numexpr          : 2.6.9
odfpy            : None
openpyxl         : 2.6.2
pandas_gbq       : None
pyarrow          : 0.11.1
pytables         : None
pytest           : 4.6.2
s3fs             : 0.2.1
scipy            : 1.2.1
sqlalchemy       : 1.3.4
tables           : 3.5.2
xarray           : 0.12.1
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.1.8

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #30173
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/construction.py [hunk 1]
```diff
@@ -574,6 +574,25 @@ def sanitize_array(
     return subarr
 
 
+def range_to_ndarray(rng: range) -> np.ndarray:
+    """
+    Cast a range object to ndarray.
+    """
+    # GH#30171 perf avoid realizing range as a list in np.array
+    try:
+        arr = np.arange(rng.start, rng.stop, rng.step, dtype="int64")
```
Comment (jreback) [inline pandas/core/construction.py:583]: isn't this very simliar to maybe_cast_to_integer_array and generate_regular_range

**Review by jreback — CHANGES_REQUESTED** [review]
rebase as well as some questions

File: pandas/core/construction.py [hunk 2]
```diff
@@ -574,6 +574,25 @@ def sanitize_array(
     return subarr
 
 
+def range_to_ndarray(rng: range) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/construction.py:577]: ideally i think there is a bunch of both dtypes.cast and core.construction that should become core/_construction/

File: pandas/core/construction.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/construction.py:583]: maybe_cast_to_integer_array no, thats the crux of #40110 

generate_regular_range conceptually, but they take different args so ill have to take a closer look

Comment (jreback) [reply pandas/core/construction.py:578]: ok as long as its on the refactor / restructure list

**Review by jreback — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**